docs/apache-airflow/concepts/operators.rst

```diff
@@ -208,3 +208,21 @@ In this case, ``order_data`` argument is passed: ``{"1001": 301.27, "1002": 433.
 Airflow uses Jinja's `NativeEnvironment <https://jinja.palletsprojects.com/en/2.11.x/nativetypes/>`_
 when ``render_template_as_native_obj`` is set to ``True``.
 With ``NativeEnvironment``, rendering a template produces a native Python type.
+
+.. _concepts:reserved-keywords:
+
+Reserved params keyword
+-----------------------
+
+In Apache Airflow 2.3.3 ``params`` variable is used during DAG serialization. Please do not use that name in third party operators.
+If you upgrade your environment and get the following error
+
+.. code-block:: python
```

Comment:
This error message is too specific to be added to the documentation. If the error message is not understandable enough, we should. update it instead of copying it to the documentation.

Reply:
Problem is placed if you provide self.params variable in your operator code. I know that is no exception handling for that issue (only class different that apache.airflow.params is covered).

You could remove code and provide more reserved words which should not be used in Operators code (as variable name for example).

Reply:
Does this problem occur with all / most constructor parameters, or only this one?

Reply:
I found it only for one Operator with internal params variable. Most of my operators works correctly after migration.

Reply:
I tihnk `params` is quite likely name that will be used by custom operators, so having a warning against this one (and the fact that it has been **just** added make it more likely for people to get the same problem.

Simply - operators that worked in 2.1 will stop working in 2.2 - which is kinda unexpected. So I think this message is useful